File: Backend/BloodDonationSupport/BloodDonationSupport.Infrastructure/Persistence/Models/Request.cs
namespace BloodDonationSupport.Infrastructure.Persistence.Models;

public partial class Request
{
    public long RequestId { get; set; }

    public long RequesterUserId { get; set; }

    public string Urgency { get; set; } = null!;

    public int BloodTypeId { get; set; }

    public int ComponentId { get; set; }

    public int QuantityUnits { get; set; }

    public DateTime? NeedBeforeUtc { get; set; }

    public long? DeliveryAddressId { get; set; }

    public string Status { get; set; } = null!;

    public string? ClinicalNotes { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }
    public double? Latitude { get; set; }  
    public double? Longitude { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual BloodType BloodType { get; set; } = null!;

    public virtual BloodComponent Component { get; set; } = null!;

    public virtual Address? DeliveryAddress { get; set; }

    public virtual ICollection<InventoryUnit> InventoryUnits { get; set; } = new List<InventoryUnit>();

    public virtual ICollection<Match> Matches { get; set; } = new List<Match>();

    public virtual ICollection<RequestEvent> RequestEvents { get; set; } = new List<RequestEvent>();

    public virtual User RequesterUser { get; set; } = null!;
}

File: Frontend/src/types/request.ts
// src/types/request.ts

export interface RegisterRequestRequest {
  requesterUserId: number;
  urgency: string; // "LOW" | "NORMAL" | "HIGH"
  bloodTypeId: number;
  componentId: number;
  quantityUnits: number;
  needBeforeUtc: string; // ISO 8601 format
  clinicalNotes: string;
  // Địa chỉ giao nhận dạng text (backend hiện hỗ trợ Option B - deliveryAddress)
  deliveryAddress: string; // fullAddress string
}

export interface Request {
  requestId: number;
  requesterUserId: number;
  urgency: string;
  bloodTypeId: number;
  componentId: number;
  quantityUnits: number;
  needBeforeUtc: string;
  deliveryAddress: string;
  clinicalNotes: string;
  status: string;
  createdAt: string;
  updatedAt?: string;
}

export interface RequestResponse {
  success: boolean;
  message?: string;
  data?: Request;
}

export interface RequestsResponse {
  success: boolean;
  message?: string | null;
  data?: Request[];
}

